### PR TITLE
Add local task executor

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

*(For example: This pull request makes user can stop workflow externally.)*

## Brief change log

*(for example:)*
  - Add task executor API 
  - Add local task executor implementation

## Verifying this change

*(Please pick either of the following options)*

This change added tests.

### CODE REVIEW

File: ai_flow/task_executor/task_executor.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        """
+        Start the task execution on worker.
+
+        :param key: Id of the task execution
+        :return: The handle of the submitted task execution, which can be used to stop the task execution
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def stop_task_execution(self, key: TaskExecutionKey, handle: str):
```
Comment (wuchaochen) [inline ai_flow/task_executor/task_executor.py:69]: handle unused.

File: ai_flow/task_executor/task_executor.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+class BaseTaskExecutor(ABC):
+
+    def __init__(self):
+        pass
+
+    def schedule_task(self, command: ScheduleTaskCommand):
+        if command.action == TaskAction.START:
+            self.start_task_execution(command.task_execution)
+        elif command.action == TaskAction.STOP:
+            self.stop_task_execution(command.task_execution)
```
Comment (wuchaochen) [inline ai_flow/task_executor/task_executor.py:54]: The handle parameter is not passed.

File: ai_flow/tests/common/util/test_process_utils.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class TestStoppableThread(StoppableThread):
+    def run(self) -> None:
+        while not self.stopped():
+            time.sleep(1)
+            print('waiting...')
+
+
+class TestLocalExecutor(unittest.TestCase):
```
Comment (wuchaochen) [inline ai_flow/tests/common/util/test_process_utils.py:32]: Rename TestLocalExecutor-> TestStoppableThread

File: ai_flow/tests/task_executor/local/test_local_task_executor.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            config_constants, 'EXECUTE_TASKS_IN_NEW_INTERPRETER', new_callable=mock.PropertyMock
+        ) as option:
+            option.return_value = True
+            self._test_execute_task()
+
+    def test_execution_with_fork(self):
+        self._test_execute_task()
+
+    def test_stop_task_execution(self):
+        pass
```
Comment (wuchaochen) [inline ai_flow/tests/task_executor/local/test_local_task_executor.py:45]: Need to add tests

File: ai_flow/common/util/process_utils.py [hunk 5]
```diff
@@ -60,14 +60,30 @@ def stop_process(pid, process_name, timeout_sec: int = 60):
...
             os.kill(pid, signal.SIGKILL)
         except Exception as e:
-            raise RuntimeError("Failed to kill {} pid: {} with SIGKILL.".format(process_name, pid)) from e
+            raise RuntimeError("Failed to kill pid: {} with SIGKILL.".format(pid)) from e
 
-    logger.info("{} pid: {} stopped".format(process_name, pid))
+    logger.info("pid: {} stopped".format(pid))
+
+
+class StoppableThread(threading.Thread):
```
Comment (wuchaochen) [inline ai_flow/common/util/process_utils.py:76]: Move to thread_util

File: ai_flow/task_executor/local/local_task_executor.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+
+class LocalTaskExecutor(BaseTaskExecutor):
+
+    def __init__(self,
+                 parallelism: int = LOCAL_TASK_EXECUTOR_PARALLELISM):
+        self.manager: Optional[SyncManager] = None
+        self.task_queue: Optional['Queue[TaskExecutionCommandType]'] = None
+        self.result_queue: Optional['Queue[TaskExecutionStatusType]'] = None
+        self.parallelism = parallelism
```
Comment (wuchaochen) [inline ai_flow/task_executor/local/local_task_executor.py:47]: parallelism->max_ parallelism

File: ai_flow/task_executor/local/local_task_executor.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        self._task_status_observer.join()
+
+    def recover(self):
+        # TODO recover state
+        pass
+
+    def _update_status(self):
+        while not threading.current_thread().stopped():
+            try:
+                key, status = self.result_queue.get(timeout=1)
```
Comment (wuchaochen) [inline ai_flow/task_executor/local/local_task_executor.py:117]: timeout is None

File: ai_flow/task_executor/local/local_task_executor.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        """
+        Do some initialization, e.g. start a new thread to observe the status
+        of all task executions and update the status to metadata backend.
+        """
+        if self.parallelism <= 0:
+            raise AIFlowException("Parallelism of LocalTaskExecutor should be a positive integer.")
+
+        self.manager = Manager()
+        self.task_queue = self.manager.Queue()
+        self.result_queue = self.manager.Queue()
```
Comment (wuchaochen) [inline ai_flow/task_executor/local/local_task_executor.py:83]: Set max_size

File: ai_flow/tests/common/util/test_process_utils.py (hunk 3 above)
Comment (jiangxin369) [reply ai_flow/tests/common/util/test_process_utils.py:32]: Renamed

File: ai_flow/common/util/process_utils.py (hunk 5 above)
Comment (jiangxin369) [reply ai_flow/common/util/process_utils.py:76]: OK

**Review by wuchaochen — APPROVED** [review]

**Pull request merged**